gradle-client/build.gradle.dcl

```diff
@@ -0,0 +1,159 @@
+                    implementation("org.jetbrains.compose.foundation:foundation:1.6.11")
+                    implementation("org.jetbrains.compose.material3:material3:1.6.11")
+                    implementation("org.jetbrains.compose.material:material-icons-extended:1.6.11")
+                    implementation("org.jetbrains.compose.ui:ui:1.6.11")
+                    implementation("org.jetbrains.compose.components:components-resources:1.6.11")
+                    implementation("org.jetbrains.compose.components:components-ui-tooling-preview:1.6.11")
+                    implementation("org.jetbrains.compose.desktop:desktop-jvm-macos-arm64:1.6.11")
+                }
+
+                testing {
+                    dependencies {
+                        // TODO:DCL Compose doesn't play well with DCL SoftwareTypes
+                        // But we can determine the value of this string at runtime and just hardcode it
+                        implementation("org.jetbrains.compose.ui:ui-test-junit4:1.6.11")
+
+                        runtimeOnly("org.junit.vintage:junit-vintage-engine:5.12.0")
+                    }
+                }
+            }
+        }
+    }
+
+    sqlDelight {
+        databases {
+            database("ApplicationDatabase") {
+                packageName = "org.gradle.client.core.database.sqldelight.generated"
+                verifyDefinitions = true
+                verifyMigrations = true
+                deriveSchemaFromMigrations = true
+                generateAsync = false
+            }
+        }
+    }
+
+    detekt {
+        source = listOf(layout.projectDirectory.dir("src/jvmMain/kotlin"), layout.projectDirectory.dir("src/jvmTest/kotlin"))
+        config = listOf(layout.projectDirectory.file("../gradle/detekt/detekt.conf"))
+        parallel = true
+    }
+
+    compose {
+        mainClass = "org.gradle.client.GradleClientMainKt"
+
+        // TODO:DCL This could use a map model when one is available in DCL
+        jvmArgs {
+            jvmArg("-Xms") {
+                value = "35m"
+            }
+            jvmArg("-Xmx") {
+                value = "128m"
+            }
+
+            // This was originally added at an inner nesting level, but it's not clear why
+            jvmArg("-splash") {
+                value = ":\"\$APPDIR/resources/splash.png\""
+            }
+        }
+
+        buildTypes {
+            release {
+                proguard {
+                    optimize = false
+                    obfuscate = false
+                    configurationFiles = listOf(layout.projectDirectory.file("proguard-desktop.pro"))
+                }
+            }
+        }
+
+        nativeDistributions {
+            // TODO:DCL Soon, we will be able to use unqualified enums in a list in DCL, but not yet
+            targetFormats = listOf("Dmg", "Msi", "Deb")
+
+            packageName = "GradleClient"
+            packageVersion = "1.1.3"
```

Comment:
Could this be a convention set in the plugin to derive it from the version above?

Reply:
Speaking of conventions, I didn't try to add any defaults in Settings yet in this project.  Is that something to do next, or does it not matter?